lib/spack/spack/database.py

```diff
@@ -355,14 +364,23 @@ def __init__(self, root, db_dir=None, upstream_dbs=None,
         # message)
         self._fail_when_missing_deps = False
 
+        self._write_transaction_builder = lk.WriteTransaction
+        self._read_transaction_builder = lk.ReadTransaction
+        self._dump_all_fields = True
+
+    def enable_buildcache_index_mode(self):
+        self._write_transaction_builder = nullcontext
+        self._read_transaction_builder = nullcontext
+        self._dump_all_fields = False
```

Comment:
I think this needs to be a context manager to swap the `self._*_transaction_builder` so that they return to the original mode after they are used.

Also, `self._*_transaction_builder` might be clearer as `self._*_transaction_impl`

Reply:
Looking now at how to implement this suggestion, thanks.